modules/cosmosBidAdapter.js

```diff
@@ -0,0 +1,398 @@
+
+      if (!utils.isEmpty(bid.params.publisherId) ||
+        !utils.isNumber(bid.params.publisherId)) {
+        utils.logError(LOG_PREFIX, 'publisherId is mandatory and must be numeric. Ad Unit: ', JSON.stringify(bid));
+        break;
+      }
+      // video bid request validation
+      if (bid.params.hasOwnProperty('video')) {
+        if (!bid.params.video.hasOwnProperty('mimes') ||
+          !utils.isArray(bid.params.video.mimes) ||
+          bid.params.video.mimes.length === 0) {
+          utils.logError(LOG_PREFIX, 'mimes are mandatory for video bid request. Ad Unit: ', JSON.stringify(bid));
+          break;
+        }
+      }
+      retval = true;
+    } while (0);
+    return retval;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   **/
+  buildRequests: function (validBidRequests, bidderRequest) {
+    if (validBidRequests.length === 0) {
+      return [];
+    }
+
+    return validBidRequests.map(bidRequest => {
+      const oRequest = spec._createRequest(bidRequest);
+      if (oRequest) {
+        spec._setGDPRParams(bidderRequest, oRequest);
+        return {
+          method: HTTP_POST,
+          url: BID_ENDPOINT,
+          data: JSON.stringify(oRequest)
+        };
+      }
+    });
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {ServerResponse} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   **/
+  interpretResponse: function (serverResponse, request) {
+    let response = serverResponse.body;
+    var bidResponses = [];
+    try {
+      if (response.seatbid) {
+        var currency = response.cur ? response.cur : DEFAULT_CURRENCY;
+        response.seatbid.forEach(seatbid => {
+          var bids = seatbid.bid ? seatbid.bid : [];
+          bids.forEach(bid => {
+            var bidResponse = {
+              requestId: bid.impid,
+              cpm: (parseFloat(bid.price) || 0).toFixed(2),
+              width: bid.w,
+              height: bid.h,
+              creativeId: bid.crid,
+              currency: currency,
+              netRevenue: DEFAULT_NET_REVENUE,
+              ttl: 300
+            };
+            if (bid.dealid) {
+              bidResponse.dealId = bid.dealid;
+            }
+
+            var req = spec._parse(request.data);
+            if (req.imp && req.imp.length > 0) {
+              req.imp.forEach(impr => {
+                if (impr.id === bid.impid) {
+                  if (impr.banner) {
+                    bidResponse.ad = bid.adm;
+                  } else {
+                    bidResponse.width = bid.hasOwnProperty('w') ? bid.w : impr.video.w;
+                    bidResponse.height = bid.hasOwnProperty('h') ? bid.h : impr.video.h;
+                    bidResponse.vastXml = bid.adm;
```

Comment:
Please add a line somewhere in this `else` block to specify the following property:
`bidResponse.mediaType = 'video';`

This will properly flag the bid as a `video` bid, which is needed for some logic later in the auction process.

Reply:
Thanks for input. Now we set `mediaType` for both banner and video response.